### ISSUE TITLE
Support Pydantic model attributes

### ISSUE THREAD

**Issue by StephenBrown2** [pawamoy/pytkdocs#9]
I'm using [Pydantic](https://github.com/samuelcolvin/pydantic) for several models in my code, and I would like mkdocstrings to handle them. The issue is, class attributes get sucked up into a `__fields__` attribute with separate `__annotations__`, and so they get filtered out by [`filter_name_out`](https://github.com/pawamoy/mkdocstrings/blob/ca82f6964e223b7d8b0a3d0e0fe4d9ea8f4296ce/src/mkdocstrings/documenter.py#L399) due to the [`global_filters`](https://github.com/pawamoy/mkdocstrings/blob/27dbdbbd0d947d67b9b618cf12ffc16ec3ab15ee/src/mkdocstrings/plugin.py#L22).

Example `__dict__.items()` on a Pydantic class with two fields (debugged from just before https://github.com/pawamoy/mkdocstrings/blob/ca82f6964e223b7d8b0a3d0e0fe4d9ea8f4296ce/src/mkdocstrings/documenter.py#L340):
```python
class_.__dict__.items(): (
    dict_items(
        [
            ('__config__', <class 'pydantic.main.Config'>),
            ('__fields__', {
                'start': ModelField(name='start', type=DateTime, required=True),
                'end': ModelField(name='end', type=DateTime, required=True)
            }),
            ('__field_defaults__', {}),
            ('__validators__', {
                'start': [<pydantic.class_validators.Validator object at 0x10257df10>],
                'end': [<pydantic.class_validators.Validator object at 0x10257df10>]
            }),
            ('__pre_root_validators__', []),
            ('__post_root_validators__', []),
            ('__schema_cache__', {}),
            ('__json_encoder__', <staticmethod object at 0x10260e978>),
            ('__custom_root_type__', False),
            ('__module__', 'mypackage.models.available_times'),
            ('__annotations__', {
                'start': <class 'pendulum.datetime.DateTime'>,
                'end': <class 'pendulum.datetime.DateTime'>
            }),
            ('__doc__', '### TimeSlot\n\n    A slot of time in which a maintenance may be scheduled.\n    '),
            ('__lt__', <function TimeSlot.__lt__ at 0x1026152f0>),
            ('parse_times', <classmethod object at 0x10260e828>),
            ('__weakref__', <attribute '__weakref__' of 'TimeSlot' objects>),
            ('__abstractmethods__', frozenset()),
            ('_abc_registry', <_weakrefset.WeakSet object at 0x10260e908>),
            ('_abc_cache', <_weakrefset.WeakSet object at 0x10260ea58>),
            ('_abc_negative_cache', <_weakrefset.WeakSet object at 0x10260eac8>),
            ('_abc_negative_cache_version', 42)
        ]
    )
)
```

A crude implementation for an older version of pydantic was put forward here: https://github.com/samuelcolvin/pydantic/issues/638 but not accepted into Pydantic proper. Perhaps mkdocstrings could take some of the logic and parse through the field attributes?

Update: I should probably also note that I started down this path by attempting to add some docstrings to the attributes as suggested by mkdocstrings docs, but then got this error:
```py
  File "/Users/step7212/git/rack/emcee/.nox/docs-3-6/lib/python3.6/site-packages/mkdocstrings/documenter.py", line 437, in node_to_annotated_names
    name = node.target.attr
AttributeError: 'Name' object has no attribute 'attr'
```
Changing `node.target.attr` to `node.target.id` ~fixed that for me~ avoided the error, but I'm not sure how relevant it is, just means I can't docstring my pydantic models (yet) since it's filtering the dunder-attrs.

**Comment by pawamoy** [comment]
@shyamd now that the refactor is done, you're free to try and come up with a PR :slightly_smiling_face:
I've made the code a bit more structured, robust and readable. I guess supporting Pydantic models would require to update the `get_class_documentation` method: https://github.com/pawamoy/pytkdocs/blob/8dfd0bbf2e73977fc1cc9528e62057f3764a5a07/src/pytkdocs/loader.py#L302

**Comment by shyamd** [comment]
So both `dataclass` and `pydantic` models have `__annotations__`. I can keep it strict to just that, or I can add in code specific to `pydantic`. Do you have any issue with adding in `pydantic` specific compatibility code?

**Comment by pawamoy** [comment]
> Do you have any issue with adding in pydantic specific compatibility code?

Not at all! This should simply be properly commented / documented :slightly_smiling_face:

**Comment by StephenBrown2** [comment]
Might want to start with the more generic approach, being able to knock out both stdlib dataclasses and Pydantic models would be a great benefit!

**Comment by pawamoy** [comment]
I think that's what @shyamd meant?

> I can keep it strict to just that, or I can add in code specific to pydantic

- Just that = at least dataclass support, maybe basic pydantic support
- add in = both dataclass and pydantic support

**Comment by pawamoy** [comment]
Hi @StephenBrown2, @demospace!

@shyamd just got their pull request merged in master! If you want to try them now, before I make a new release, you can simply `pip install git+https://github.com/pawamoy/pytkdocs` in your repository to see if Pydantic and dataclass attributes are correctly picked up and rendered by mkdocstrings :slightly_smiling_face:

**Comment by StephenBrown2** [comment]
So I tried real hard, but it seems that it's not working in my main project, but it does work on a minimal demo project. Probably something on my end, but it's a good start!

**Comment by pawamoy** [comment]
About dataclasses fields not having a description, yeah, we decided to do that in another PR, because it needs a refactoring of the attributes parser. I'll go ahead and put this in the "to release" column :slightly_smiling_face:

**Comment by pawamoy** [comment]
Released in v0.4.0. Please open new issues for any encountered problem with this feature!
